### DISCUSSION TITLE
Add swipe to mark read option

### DISCUSSION THREAD

**Discussion by M7amma** [jocmp/capyreader#273] [Feature Requests]
Please add option to allow deleting feed item by swiping on it from left to right. (This is the most needed option for me to make it perfect). Thanks 🙏

**Comment by jocmp** [comment]
@M7amma I'd like to add this too. Here's a very early prototype of what I have in mind. 

https://github.com/user-attachments/assets/ef2bd589-32ac-4782-ac81-b08ef8d3c86c

**Reply by M7amma** [reply]
Looks very promising now. I wish to see it in the next version. Thank you so much for that.

**Comment by ConEstArthur** [comment]
@jocmp Hey, Joseph..

I use Newsflash on my pc. It has the button to manually clear all the articles. 
It'd be nice to have the same in Capy Reader as well.

I know you have added an option to automatically clear them after 3 months. It'd be more helpful for me if you'd give an option to do it manually ?

**Reply by jocmp** [reply]
hey @ConEstArthur - I can investigate how they do it. do you have a screenshot or a link to the app? There are a few Google results for Newsflash.

**Reply by ConEstArthur** [reply]
Sure.
Here you go -
https://gitlab.com/news-flash/news_flash_gtk/

**Reply by ConEstArthur** [reply]
I may have misinterpreted this option. I apologize for that. 
The option given in "newsflash" rebuilds database files not necessarily clears the feed articles.
However a manual option would be nice I think.

![image](https://github.com/user-attachments/assets/2f2ee01a-6e92-40b2-b084-3b3a3a479322)

**Reply by jocmp** [reply]
Thanks for this. I think in that case what I'd do is make the "Auto Delete" configurable to 1 week, 1 month, 2 months, etc along with an option to clear out all articles with a "Clean" button. 

Capy Reader uses a SQLite database to store articles and feeds so there isn't as big of a need to rebuild a file to my knowledge.

**Reply by ConEstArthur** [reply]
I completely missed the mark here. Actually Read You app has this option marked as "Clear All Articles"

Good option for a fresh start :)

<img src="https://github.com/user-attachments/assets/b9881a49-2ae7-47b7-b972-6209e9d42a4e" width="360" >

**Reply by jocmp** [reply]
Tracking the feature here. I'm working on a new settings page so it'll take some time before I get to this feature

- https://github.com/jocmp/capyreader/issues/310

**Comment by jocmp** [comment]
Swipe to mark read is available as of [2024.09.1053](https://github.com/jocmp/capyreader/releases/tag/2024.09.1053-dev). You can configure it to swipe to favorite or disable it completely in settings. Closing for now.
